File: updater/src/main/java/identifiers/MilliClock.java
package identifiers;

import utility.ClassWrapper;
import utility.DependsOn;

@DependsOn(Clock.class)
public class MilliClock extends AbstractIdentifier {
    @Override
    public boolean isMatch(ClassWrapper classNode) {
        return classNode.isSuperClassEquals("Clock") &&
                classNode.getInstanceFields().size() > 1;
    }
}

File: find-names/src/main/java/mapper/identifiers/classes/MilliClock.kt
package mapper.identifiers.classes

import mapper.abstractclasses.IdentityMapper
import mapper.annotations.DependsOn
import mapper.annotations.MethodParameters
import mapper.predicateutilities.and
import mapper.predicateutilities.predicateOf
import mapper.wrappers.ClassWrapper
import mapper.wrappers.MethodWrapper
import org.objectweb.asm.Type

@DependsOn(Clock::class)
class MilliClock : IdentityMapper.Class() {

    override val predicate = predicateOf<ClassWrapper> { it.superType == type<Clock>() }
            .and { it.instanceFields.size > 1 }

    @MethodParameters()
    class mark : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == Type.VOID_TYPE }
    }

    @MethodParameters("cycleMs", "minSleepMs")
    class wait : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == Type.INT_TYPE }
    }
}